src/structures/AutocompleteInteraction.js

```diff
@@ -0,0 +1,114 @@
+'use strict';
+
+const CommandInteractionOptionResolver = require('./CommandInteractionOptionResolver');
+const Interaction = require('./Interaction');
+const { InteractionResponseTypes, ApplicationCommandOptionTypes } = require('../util/Constants');
+
+/**
+ * Represents an autocomplete interaction.
+ * @extends {Interaction}
+ */
+class AutocompleteInteraction extends Interaction {
+  constructor(client, data) {
+    super(client, data);
+
+    /**
+     * The channel this interaction was sent in
+     * @type {?TextBasedChannels}
+     * @name AutocompleteInteraction#channel
+     * @readonly
+     */
```

Comment:
This is redundant isn't it? It's identical to the type in the base class:
https://github.com/discordjs/discord.js/blob/34b2ad0d8ee39eb9cc964a4334a803b2ab48cae5/src/structures/Interaction.js#L97-L101

Reply:
If it's redundant, wouldn't BaseCommandInteraction also be doing something redundant since it also has it despite it already being in the base class?

Reply:
Probably